src/prop/prop_proof_manager.cpp

```diff
@@ -274,6 +452,98 @@ Node PropPfManager::normalizeAndRegister(TNode clauseNode,
 
 LazyCDProof* PropPfManager::getCnfProof() { return &d_proof; }
 
+void PropPfManager::getProofInternal(CDProof* cdp)
+{
+  // This method is called when the SAT solver did not generate a fully self
+  // contained ProofNode proving false. This method adds a step to cdp
+  // based on a set of computed assumptions, possibly relying on the internal
+  // proof.
+  NodeManager* nm = NodeManager::currentNM();
+  Node falsen = nm->mkConst(false);
+  std::vector<Node> clauses;
+  // deduplicate assumptions
+  Trace("cnf-input") << "#assumptions=" << d_assumptions.size() << std::endl;
+  std::vector<Node> minAssumptions = getMinimizedAssumptions();
+  if (minAssumptions.size() == 1 && minAssumptions[0] == falsen)
+  {
+    // if false exists, no proof is necessary
+    return;
+  }
+  std::unordered_set<Node> cset(minAssumptions.begin(), minAssumptions.end());
+  Trace("cnf-input") << "#assumptions (min)=" << cset.size() << std::endl;
+  std::vector<Node> inputs = getInputClauses();
+  Trace("cnf-input") << "#input=" << inputs.size() << std::endl;
+  std::vector<Node> lemmas = getLemmaClauses();
+  Trace("cnf-input") << "#lemmas=" << lemmas.size() << std::endl;
+  cset.insert(inputs.begin(), inputs.end());
+  cset.insert(lemmas.begin(), lemmas.end());
+
+  // Otherwise, we will dump a DIMACS. The proof further depends on the
+  // mode, which we handle below.
+  std::stringstream dinputFile;
+  dinputFile << options().driver.filename << ".drat_input.cnf";
+  // the stream which stores the DIMACS of the computed clauses
```

Comment:
at this point it's confusing why do this

Reply:
You mean why do we dump the dimacs to file?  This is what `SAT_EXTERNAL_PROVE` is based on.

Reply:
Looking into this, we use `dout` no matter what, since even if we don't minimize we include all clauses by default, which is written on `dout`.

Reply:
I see, sounds good